chaco/tools/range_selection.py

```diff
@@ -679,15 +679,15 @@ def _determine_axis(self):
             else:
                 return 0
 
-    def __mapper_changed(self):
+    def __mapper_changed(self, event):
         self.deselect()
         return
 
     def _axis_changed(self, old, new):
         if old is not None:
-            self.plot.on_trait_change(self.__mapper_changed,
+            self.plot.observe(self.__mapper_changed,
                                       old + "_mapper", remove=True)
         if new is not None:
-            self.plot.on_trait_change(self.__mapper_changed,
-                                      old + "_mapper", remove=True)
```

Comment:
This previous code seems like it was a copy paste error...
Butt the updated code is still not right ad we have a case in the test suite where old is non None, but its listener never was hooked up initially